### DISCUSSION TITLE
Glyphicons Halflings font license terms?

### DISCUSSION THREAD

**Discussion by KalleOlaviNiemitalo** [dotnet/docfx#9578] [Q&A]
The `docfx` .NET tool package includes the Glyphicons Halflings font and copies it to the generated web site.  How is this font licensed when shipped in DocFX; does `<license type="expression">MIT</license>` in docfx.nuspec cover it too?

From <https://www.glyphicons.com/sets/halflings/>, it seems the author wants $35 per company and does not allow sublicensing, but <https://getbootstrap.com/docs/3.4/components/> says "their creator has made them available for Bootstrap free of cost".   In <https://github.com/twbs/bootstrap/issues/3942#issuecomment-12557565>, the author posted a link to <http://glyphicons.com/glyphicons-licenses/>, but that page no longer exists.

**Comment by filzrev** [comment]
> the author posted a link to http://glyphicons.com/glyphicons-licenses/, but that page no longer 

It seems there is `older license for previous version` link at linked issue.
https://glyphicons.com/old/license.html#old-halflings-bootstrapists.

Currently docfx use these fonts when using `default` and `statictoc` template. (That use bootstrap `3.4.1`)
And bootstrap's license notice is included at `docfx.vendor.min.css` file.
So it seems no problems for generated site by docfx.

For `NuGet` distributed packages.
There is `THIRD-PARTY-NOTICES.TXT` in bin folder.
But client-side library related license notices are not included.
It might be better to include these license notice files also.

**Reply by KalleOlaviNiemitalo** [reply]
DocFX seems to be using [Bootstrap 3.4.1](https://www.npmjs.com/package/bootstrap/v/3.4.1) in the "default", "pdf.default", and "statictoc" templates; but [Bootstrap 5.3.2](https://www.npmjs.com/package/bootstrap/v/5.3.2) in the "modern" template.  That's what I find when I grep for those version numbers in the docfx tool package.  Bootstrap 5.3.2 has no fonts in the NPM package and no dependencies either.  So if the older templates are ever eliminated, then Glyphicons Halflings will no longer be available and has to be replaced, I think.

**Reply by yufeih** [reply]
The modern template uses `bootstrap-icons` instead.

**Reply by KalleOlaviNiemitalo** [reply]
🤔 When I built a PDF for a site using the "modern" template, a subset of the Glyphicons Halflings font became embedded in the PDF.  I wonder what part of the HTML still used that font.

**Comment by KalleOlaviNiemitalo** [comment]
# Context

I was setting up CSS for a custom alert type (<https://github.com/dotnet/docfx/pull/8735>).  The DocFX default styling applies the Glyphicons Halflings font to the glyph before the alert title:

<https://github.com/dotnet/docfx/blob/0d56f79e0817e99029aabb43f473685da7e202f6/templates/default/styles/docfx.css#L918-L928>

I then searched the Web for a table of Glyphicons Halflings characters in order to pick a suitable one.  The first hit was <https://www.glyphicons.com/sets/halflings/>, which does have a table of characters, but it also advertises the paid license, which made me suspect that I might not have the correct license for the font.
